cwl/src/main/scala/cwl/package.scala

```diff
@@ -86,5 +86,20 @@ package object cwl extends TypeAliases {
       case Cwl.Workflow(w) => w.womExecutable(inputsFile)
       case Cwl.CommandLineTool(clt) => clt.womExecutable(inputsFile)
     }
+
+    def requiredInputs: Map[String, WomType] = cwl match {
+      case Cwl.Workflow(w) => selectWomTypeInputs(w.inputs collect {
+        case i if i.`type`.isDefined => FullyQualifiedName(i.id).id -> i.`type`.get
+      })
+      case Cwl.CommandLineTool(clt) => selectWomTypeInputs(clt.inputs collect {
+        case i if i.`type`.isDefined => FullyQualifiedName(i.id).id -> i.`type`.get
+      })
+    }
```

Comment:
is the code duplication suggesting a common supertype? 🙂

Reply:
Perhaps, I'll let @danbills look into that. As far as I can tell, `InputParameter` and `CommandInputParameter` are the same thing

Reply:
they are indeed!  as well as all the `*Schema` classes too it seems.  I think we could de-duplicate broadly if we feel confident that CWL would not diverge in the future.  It would be nice to know why they have distinct names in the spec.

Reply:
IIRC originally there was just a single entity and then when it became clear that workflows and individual tools have slightly different needs they diverged. The other thing was that at the time, at least from my observation, that community was dominated by people who think/thought in terms of single tools and not workflows, despite the W in the acronym.